pkg/metrics/statsd/statsd.go

```diff
@@ -0,0 +1,152 @@
+package statsd
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+// statsd implementation for https://genevamondocs.azurewebsites.net/collect/references/statsdref.html
+
+// Now stubs time.Now
+var Now = time.Now
+
+var (
+	defaultSocket = "/tmp/mdm_statsd.socket"
+)
+
+// Statsd defines internal statsd client
+// It should be cloned, but not modified
+type Statsd struct {
+	account   string
+	namespace string
+	dims      map[string]string
+
+	socket string
+	conn   net.Conn
+	env    env.Interface
+	mu     sync.Mutex
+
+	// for testing
+	Muted bool
+}
+
+// New method to initialize udp connection
+func New(ctx context.Context, log *logrus.Entry, _env env.Interface) (metrics.Interface, error) {
+	// defaults- dev
+	config := &Statsd{
+		account:   os.Getenv("METRICS_ACCOUNT"),
+		namespace: os.Getenv("METRICS_NAMESPACE"),
+		dims: map[string]string{
+			"location": _env.Location(),
+		},
+		socket: defaultSocket,
+	}
+
+	var err error
+	config.conn, err = net.Dial("unix", config.socket)
```

Comment:
why not hard code to "mdm_statsd.socket" (i.e. in the current working directory?) (and add to .gitignore)